File: app/src/main/java/com/ericho/coupleshare/mvp/LoginContract.java
package com.ericho.coupleshare.mvp;

import android.support.annotation.NonNull;
import android.text.TextWatcher;
import android.widget.EditText;

/**
 * Created by steve_000 on 12/6/2017.
 * for project CoupleShare
 * package name com.ericho.coupleshare.mvp
 */

public interface LoginContract {
    interface View extends BaseView<Presenter>{

        void showLoadingIndicator(boolean active);

        void showLoginSuccess();

        void showLoginFailure(String errorMessage);

        void showRegisterPage();

        void showChangeServerAddressUi();

        void showMoreUi();

        void showLoginButtonState(boolean enable);

        void resetLoginField();
    }

    interface Presenter extends BasePresenter{

        void login(String username,String password);

        void reset();

        void requestRegisterPage();

        void registerOnTextChangeListener(@NonNull EditText usernameEditText,
                                          @NonNull EditText passwordEditText);

        void displayChangeServerAddressUi();

        void displayMoreUi();

        void destroy();
    }
}

File: app/src/main/java/com/ericho/coupleshare/mvp/LoginContract.kt
package com.ericho.coupleshare.mvp

import android.widget.Button
import android.widget.EditText

/**
 * Created by steve_000 on 8/7/2017.
 * for project CoupleShare
 * package name com.ericho.coupleshare.mvp
 */
interface LoginContract {
    interface View : BaseView<Presenter> {

        fun showLoadingIndicator(active: Boolean)

        fun showLoginSuccess()

        fun showLoginFailure(errorMessage: String?)

        fun showRegisterPage()

        fun showChangeServerAddressUi()

        fun showMoreUi()

        fun showLoginButtonState(enable: Boolean)

        fun resetLoginField()
    }

    interface Presenter : BasePresenter {

        fun login(username: String, password: String)

        fun reset()

        fun requestRegisterPage()

        fun registerOnTextChangeListener(usernameEditText: EditText,
                                         passwordEditText: EditText,
                                        loginBtn:Button)

        fun displayChangeServerAddressUi()

        fun displayMoreUi()

        fun destroy()
    }
}